app/static/js/sidebar.js

```diff
@@ -381,12 +397,36 @@ function adapt_view_to_state_level(previous_level) {
     zoomToSelectedState();
 
     // Trigger the filter function so that the selected state geojson does not hide the clusters
-    nigeria_states_geojson.clearLayers()
-    nigeria_states_geojson.addData(nigeria_states_simplified)
-  }
+    nigeria_states_geojson.clearLayers();
+    nigeria_states_geojson.addData(nigeria_states_simplified);
+  };
+  if (previous_level == "village" && trigger == "button") {
+    zoomToSelectedState(newlySelected=false);
+  };
 };
 
-function adapt_view_to_village_level() {
+function adapt_view_to_village_level(previous_level, trigger) {
+
+  // click on the village level button from national level
+  if (previous_level == "national" && trigger == "button"){
+      // select a random state which has off-grid clusters
+      hasCluster = ""
+      while (hasCluster == ""){
+        selectedState = statesList[Math.floor(Math.random()*statesList.length)]
+        hasCluster = OGClusterLayers[selectedState];
+      };
+      // Update the states menu list
+      document.getElementById("stateSelect").value = selectedState;
+  };
+  if ((previous_level == "national" || previous_level == "state") && trigger == "button"){
+    //TODO: pick a random cluster among the large ones and display it
+     $.post({
+        url: "/filter-cluster",
+        dataType: "json",
+        data: {"state_name": selectedState},
+        success: function(data){console.log(data);},
```

Comment:
`data` contains the info about the cluster from the view, with that how could I fly to the village @christian-rli ? You can take over the rest of this PR (basically replacing `console.log(data)` by the code which fly to the village).

Reply:
OK. I wanted to start the website, but after installing  flask_wtf I got the following error:

File "../index.py", line 1, in <module>
    from app import create_app
  File "/home/christianh/lokal/240_NESP2/NESP2/app/__init__.py", line 7, in <module>
    from .database import get_random_village
ModuleNotFoundError: No module named 'app.database'

Reply:
Oups I need to add his file

Reply:
Meanwhile do pip install -r requirements.txt